File: 29_divide_two_integers/pramoda/solution.py
class Solution:
    def divide(self, dividend: int, divisor: int) -> int:
        MAX_POS = 2_147_483_647
        MAX_NEG = -2_147_483_648

        if dividend == MAX_NEG and divisor == -1:
            return MAX_POS
        
        q = int(dividend/divisor)
        return q

s = Solution()
print(s.reverse(-2147483648))

File: 29_divide_two_integers/pramoda/solution.c
#include <stdio.h>
#include <stdlib.h>
#include <limits.h>

int divide(int dividend, int divisor) {
    if (dividend == INT_MIN && divisor == -1) {
        return INT_MAX;
    }
    return dividend / divisor;
}

int main() {
    
    printf("%d", divide(-2147483648, -1));

    return 0;
}